components/Map/Map.js

```diff
@@ -1,28 +1,196 @@
+  const [routeCoords, setRouteCoords] = useState([]);
+  const [calculated, setCalculated] = useState(false);
+  const [searchResults, setSearchResults] = useState([]);
+  const [inputPlaceholder, setInputPlaceholder] = useState("Search for places");
 
   mapboxgl.accessToken = accessToken;
 
+  //Map and Markers
   useEffect(() => {
     map.current = new mapboxgl.Map({
       container: mapContainer.current,
       style: "mapbox://styles/mapbox/streets-v12",
-      center: [lng, lat],
-      zoom: zoom,
+      center: [10.000654, 53.550341],
+      zoom: 9,
+    });
+
+    map.current.on("click", function (e) {
+      const markerCoords = [e.lngLat.lng, e.lngLat.lat];
+      setRouteCoords((prevCoords) => [...prevCoords, markerCoords]);
+      const marker = new mapboxgl.Marker()
+        .setLngLat(markerCoords)
+        .addTo(map.current);
+
+      setMarkers((prevMarkers) => [...prevMarkers, marker]);
     });
   }, []);
 
-  useEffect(() => {
-    if (center) {
-      map.current.flyTo({ center: center, zoom: 12 });
+  // Format the data for the Map Directions query
+  function updateRoute() {
+    const profile = "cycling";
+    // Format the coordinates
+    const newCoords = routeCoords.join(";");
+
+    getMatch(
+      `https://api.mapbox.com/directions/v5/mapbox/${profile}/${newCoords}?geometries=geojson&language=de&overview=full&steps=true&access_token=${mapboxgl.accessToken}`
+    );
+  }
+
+  // Make a Map Matching request
+  async function getMatch(url) {
+    setIsLoading(true);
+
+    const response = await fetch(url);
+    const data = await response.json();
+    setNewRouteData(data);
+
+    // Handle errors
+    if (!response) {
+      alert(
+        `${data.code} - ${data.message}.\n\nFor more information: https://docs.mapbox.com/api/navigation/map-matching/#map-matching-api-errors`
+      );
+      return;
+    }
+
+    // Get the coordinates from the response
+    const coords = data.routes[0].geometry;
+
+    // Draw the route on the map
+    addRoute(coords);
+
+    setCalculated(true);
+    setIsLoading(false);
+  }
+
+  // Draw Route on the Map
+  function addRoute(coords) {
+    // If a route is already loaded, remove it
+    if (map.current.getSource("route")) {
+      map.current.removeLayer("route");
+      map.current.removeSource("route");
     }
-  }, [center]);
 
-  return <StyledMap ref={mapContainer} />;
+    // Add a new layer to the map
+    map.current.addLayer({
+      id: "route",
+      type: "line",
+      source: {
+        type: "geojson",
+        data: {
+          type: "Feature",
+          properties: {},
+          geometry: coords,
+        },
+      },
+      layout: {
+        "line-join": "round",
+        "line-cap": "round",
+      },
+      paint: {
+        "line-color": "#03AA46",
+        "line-width": 8,
+        "line-opacity": 0.8,
+      },
+    });
+  }
+
+  // Make a Geocoding request
+  const getNewDestination = async (url) => {
```

Comment:
you switch between function declarations and arrow functions.
I would advise consistency (the same goes for all upcoming functions):

```suggestion
  async function getNewDestination(url) {
```